app/src/main/java/com/zulip/android/networking/GravatarAsyncFetchTask.java

```diff
@@ -47,7 +48,7 @@ public static URL sizedURL(Context context, String url, float dpSize) {
         float px = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP,
                 dpSize, r.getDisplayMetrics());
         try {
-            return new URL(url + "&s=" + px);
+            return new URL(ZulipApp.get().getServerHostUri() + url + "&s=" + px);
```

Comment:
Updated :+1: Now it handles both cases.
![18575338_796557247204752_1421320719_o](https://cloud.githubusercontent.com/assets/18511177/26124439/087702b4-3a9c-11e7-89b1-976623d237e8.jpg)
![18519306_796557150538095_1647942643_o](https://cloud.githubusercontent.com/assets/18511177/26124442/095c1dc2-3a9c-11e7-803e-aa25486ff000.jpg)